lib/Transforms/Scalar/DxilLoopUnroll.cpp

```diff
@@ -522,8 +523,16 @@ static bool BreakUpArrayAllocas(bool AllowOOBIndex, IteratorT ItBegin, IteratorT
 
     GEPs.clear(); // Re-use array
     for (User *U : AI->users()) {
-      // Try to set all GEP operands to constant
-      if (GEPOperator *GEP = dyn_cast<GEPOperator>(U)) {
+      // Ignore lifetime intrinsics
+      if (BitCastInst *BCI = dyn_cast<BitCastInst>(U)) {
+        DXASSERT(onlyUsedByLifetimeMarkers(BCI),
+                 "expected bitcast to only be used by lifetime intrinsics");
```

Comment:
Is there no possibility that bitcast for memcpy could be present at this point, or something else?